runhouse/resources/hardware/cluster.py

```diff
@@ -703,10 +717,12 @@ def restart_server(
         logger.info(f"Restarting Runhouse API server on {self.name}.")
 
         if resync_rh:
-            self._sync_runhouse_to_cluster(_install_url=_rh_install_url)
+            if not self.address:
```

Comment:
Should we add a comment on what is the logical condition here e.g. in what cases would self.address be None?